src/Agent.Listener/Agent.cs

```diff
@@ -282,13 +282,14 @@ private async Task<int> RunAsync(AgentSettings settings, bool runOnce = false)
                 try
                 {
                     var notification = HostContext.GetService<IJobNotification>();
+
                     if (!String.IsNullOrEmpty(settings.NotificationSocketAddress))
                     {
-                        notification.StartClient(settings.NotificationSocketAddress);
+                        notification.StartClient(settings.NotificationSocketAddress, settings.MonitorPort);
                     }
                     else
                     {
-                        notification.StartClient(settings.NotificationPipeName, HostContext.AgentShutdownToken);
+                        notification.StartClient(settings.NotificationPipeName, settings.MonitorPort, HostContext.AgentShutdownToken);
```

Comment:
check with MMS team, see whether they still using pipe.

Reply:
I'll get that done outside the PR scope. If they don't use pipes, they should probably own cleaning up this code?